reinforcement_learning/rl_cartpole_ray/Dockerfile

```diff
@@ -0,0 +1,50 @@
+ARG CPU_OR_GPU
+ARG AWS_REGION
+FROM 462105765813.dkr.ecr.${AWS_REGION}.amazonaws.com/sagemaker-rl-ray-container:ray-0.8.2-tf-${CPU_OR_GPU}-py36
+
+WORKDIR /opt/ml
+
+# Roboschool dependencies
```

Comment:
Why would we need to build our own docker here. The public RL container image should suffice?

Reply:
So I tried running the vanilla Docker image without creating a new one, and I seem to be running into this error:
```

  | 2020-07-16T14:56:47.502-07:00 | Traceback (most recent call last):  File "/usr/local/lib/python3.6/dist-packages/ray/tune/trial_runner.py", line 459, in _process_trial    result = self.trial_executor.fetch_result(trial)  File "/usr/local/lib/python3.6/dist-packages/ray/tune/ray_trial_executor.py", line 377, in fetch_result    result = ray.get(trial_future[0], DEFAULT_GET_TIMEOUT)  File "/usr/local/lib/python3.6/dist-packages/ray/worker.py", line 1504, in get    raise value.as_instanceof_cause()
-- | -- | --
  | 2020-07-16T14:56:47.502-07:00 | ray.exceptions.RayTaskError(AttributeError): #033[36mray::PPO.train()#033[39m (pid=106, ip=10.0.188.140)  File "python/ray/_raylet.pyx", line 452, in ray._raylet.execute_task  File "python/ray/_raylet.pyx", line 430, in ray._raylet.execute_task.function_executor  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/agents/trainer.py", line 494, in train    raise e  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/agents/trainer.py", line 483, in train    result = Trainable.train(self)  File "/usr/local/lib/python3.6/dist-packages/ray/tune/trainable.py", line 254, in train    result = self._train()  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/agents/trainer_template.py", line 133, in _train    fetches = self.optimizer.step()  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/optimizers/multi_gpu_optimizer.py", line 137, in step    self.num_envs_per_worker, self.train_batch_size)  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/optimizers/rollout.py", line 25, in collect_samples    next_sample = ray_get_and_free(fut_sample)  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/utils/memory.py", line 29, in ray_get_and_free    result = ray.get(object_ids)
  | 2020-07-16T14:56:47.503-07:00 | ray.exceptions.RayTaskError(AttributeError): #033[36mray::RolloutWorker.sample()#033[39m (pid=104, ip=10.0.188.140)  File "python/ray/_raylet.pyx", line 452, in ray._raylet.execute_task  File "python/ray/_raylet.pyx", line 430, in ray._raylet.execute_task.function_executor  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/evaluation/rollout_worker.py", line 488, in sample    batches = [self.input_reader.next()]  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/evaluation/sampler.py", line 52, in next    batches = [self.get_data()]  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/evaluation/sampler.py", line 95, in get_data    item = next(self.rollout_provider)  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/evaluation/sampler.py", line 301, in _env_runner    base_env.poll()  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/env/base_env.py", line 308, in poll    self.new_obs = self.vector_env.vector_reset()  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/env/vector_env.py", line 96, in vector_reset    return [e.reset() for e in self.envs]  File "/usr/local/lib/python3.6/dist-packages/ray/rllib/env/vector_env.py", line 96, in <listcomp>    return [e.reset() for e in self.envs]  File "/usr/local/lib/python3.6/dist-packages/gym/wrappers/monitor.py", line 39, in reset    self._after_reset(observation)  File "/usr/local/lib/python3.6/dist-packages/gym/wrappers/monitor.py", line 188, in _after_reset    self.reset_video_recorder()  File "/usr/local/lib/python3.6/dist-packages/gym/wrappers/monitor.py", line 209, in reset_video_recorder    self.video_recorder.capture_frame()  File "/usr/local/lib/python3.6/dist-packages/gym/wrappers/monitoring/video_recorder.py", line 101, in capture_frame    frame = self.env.render(mode=render_mode)  File "/usr/local/lib/python3.6/dist-packages/gym/core.py", line 249, in render    return self.env.render(mode, **kwargs)  File "/usr/local/lib/python3.6/dist-packages/gym/envs/classic_control/cartpole.py", line 188, in render    return self.viewer.render(return_rgb_array = mode=='rgb_array')  File "/usr/local/lib/python3.6/dist-packages/gym/envs/classic_control/rendering.py", line 105, in render    arr = np.frombuffer(image_data.data, dtype=np.uint8)
  | 2020-07-16T14:56:47.503-07:00 | AttributeError: 'ImageData' object has no attribute 'data'


```
I resolved this by forcing install of the `pyglet` library to the version 1.3.2. But there is a different version of `pyglet` installed in the image, so I think we have two options here:

1. Create a new docker image and force install `pyglet==1.3.2` ourselves.
2. Update the existing docker image for Sagemaker RL so it uses `pyglet 1.3.2`. 

I don't know if there will be any side-effects if we choose the 2nd option, so IMO creating a new docker image and overriding with the correct version of this lib would be a safer option.

Reply:
I reproduced the same error (`'ImageData' object has no attribute 'data'`) by using the latest public RL container Tensorflow image (462105765813.dkr.ecr.us-west-2.amazonaws.com/sagemaker-rl-ray-container:ray-0.8.5-tf-gpu-py36).